server/controllers/bookingRequest.js

```diff
@@ -0,0 +1,121 @@
+const asyncHandler = require("express-async-handler");
+const User = require("../models/User");
+const BookingRequest = require("../models/BookingRequest");
+
+// @route POST /bookingrequests/create
+// @desc Create a new request
+// @access Private
+
+exports.createBookingRequest = asyncHandler(async (req, res, next) => {
+  try {
+    const userId = req.user.id;
+    const user = await User.findById(userId);
```

Comment:
Optional: Good thinking, you can add this check to the `protect` middleware though, that will avoid the need to rewrite it within multiple controllers :)

Reply:
resolved by 6d8d8e9593db44e0297dc0d37d579da5eabf541f